### ISSUE TITLE
C++: Remove Templating from Libs/Optimize

### ISSUE THREAD

**Issue by HeavenlyBerserker** [SCIInstitute/ShapeWorks#739]
Templating is not used, so it should be removed to facilitate new functions and classes.
Alan: Trying to change anything causes issues for other people who work on Libs/Optimize when merging.

(long term goal)

**Comment by sheryjoe** [comment]
@akenmorris @HeavenlyBerserker Is this planned for the next release?

**Comment by akenmorris** [comment]
No, it's a long term goal.  We're part way there.  I wouldn't tie it to any release as it does not affect users or the computation.  It will speed development when done though.

**Comment by akenmorris** [comment]
Getting there.  Have removed lots of the useless templates.

**Comment by HeavenlyBerserker** [comment]
Remaining templates are in

- [ ] Container/GenericContainer and inherited classes
- [ ] Domain/ImageDomain and inherited classes
- [ ] Domain/VtkMeshWrapper
- [ ] Neighborhood/PowerOfTwoPointTree
- [ ] Utils/ObjectReader and Writer
- [ ] Utils/ParticleGaussianModeWriter

Not all of them need to have templates removed, I think. I'm looking into Domain/VtkMeshWrapper through issue #1408.
